File: move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ip = 0
        for n in nums:
            if n != 0:
                nums[ip] = n
                ip += 1
        for i in range(ip, len(nums)):
            nums[i] = 0
        
        

File: move-zeroes.go
func moveZeroes(nums []int) {
	ip := 0 // insert index/position (ip)
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[ip] = nums[i]
			ip += 1
		}
	}
	for ; ip < len(nums); ip++ {
		nums[ip] = 0
	}
}